include/oneapi/dpl/pstl/parallel_backend_utils.h

```diff
@@ -186,7 +188,7 @@ __set_intersection_construct(_ForwardIterator1 __first1, _ForwardIterator1 __las
         {
             if (!__comp(*__first2, *__first1))
             {
-                ::new (::std::addressof(*__result)) _Tp(*__first1);
+                ::new (::std::addressof(*__result)) _Tp(*__select_copy(__first1, __first2));
```

Comment:
Since there are just two cases, would it be clearer to just pass a `std::true_type`, `std::false_type`, or just a boolean as a template parameter for whether we copy from the first or second iterator instead of taking a lambda argument?

Reply:
I considered this, but I thought this may be more descriptive / less disruptive to the code.  

Do you think it would be better to have `if constexpr` in the body of `_set_intersection_construct`? and just a 
`/*CopyFromFirstSet = */ ::std::true_type` here?

I don't have a strong opinion either way.

Reply:
I looked at this a little more. Since there is no ternary constexpr, its ugly and replicates code to use `if constexpr` in the body in this context.  You'd need to make a pair of copy selection function helpers which would end up pretty similar to this version with lambdas.

Reply:
Okay, I think the current way you've implemented is best then.

Reply:
I found a decent way to do this in `_set_intersection_construct` which seems slightly better.

Reply:
This new solution seems the best to me.